cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -92,6 +93,10 @@ func enforceRequirements(flags *applyPlanFlags, dryRun bool, newK8sVersion strin
 		return nil, nil, nil, errors.Wrapf(err, "couldn't create a Kubernetes client from file %q", flags.kubeConfigPath)
 	}
 
+	if err := runCoreDNSpreflightChecks(client, ignorePreflightErrorsSet); err != nil {
```

Comment:
the upgrade checks should have a single entry point: `runPreflightChecks`
the call to `runCoreDNSpreflightChecks()` should be moved there.
`client` needs to be passed to `runPreflightChecks`.